numpy/core/src/multiarray/ctors.c

```diff
@@ -1315,27 +1356,73 @@ _array_from_buffer_3118(PyObject *memoryview)
     npy_intp shape[NPY_MAXDIMS], strides[NPY_MAXDIMS];
 
     view = PyMemoryView_GET_BUFFER(memoryview);
```

Comment:
Is is safe to assume that `memoryview` is actually a memoryview?

Reply:
Yes, because this function is only ever called with one, I think